cypress/e2e/tests/pages/user_menu/preferences.spec.ts

```diff
@@ -0,0 +1,368 @@
+    Deselect the checkbox and verify 'View in API' is hidden
+    */
+    const clusterRepoEndpoint = 'c/_/manager/catalog.cattle.io.clusterrepo';
+
+    PagePo.goTo(endpoint);
+    const checkboxLabel = 'Enable "View in API"';
+
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isChecked();
+
+    cy.intercept('GET', '/v1/catalog.cattle.io.clusterrepos').as('clusterRepos');
+    PagePo.goTo(clusterRepoEndpoint);
+    cy.wait('@clusterRepos').its('response.statusCode').should('eq', 200);
+    repoList.actionMenu('Partners').getMenuItem('View in API').should('exist');
+
+    PagePo.goTo(endpoint);
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isUnchecked();
+
+    PagePo.goTo(clusterRepoEndpoint);
+    cy.wait('@clusterRepos').its('response.statusCode').should('eq', 200);
+    repoList.actionMenu('Partners').getMenuItem('View in API').should('not.exist');
+  });
+
+  it('Can select Show system Namespaces managed by Rancher (not intended for editing or deletion)', () => {
+    /*
+    Select checkbox option and verify state is preserved after logout/login
+    */
+    PagePo.goTo(endpoint);
+    const checkboxLabel = 'Show system Namespaces managed by Rancher (not intended for editing or deletion)';
+
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isChecked();
+    userMenu.open();
+    userMenu.checkOpen();
+    userMenu.clickMenuLink('Log Out');
+    cy.login();
+    PagePo.goTo(endpoint);
+    prefPage.checkbox(checkboxLabel).isChecked();
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isUnchecked();
+  });
+
+  it('Can select Enable Dark/Light Theme keyboard shortcut toggle (shift+T)', () => {
+    /*
+    Select checkbox option and verify state is preserved after logout/login
+    */
+    PagePo.goTo(endpoint);
+    const checkboxLabel = 'Enable Dark/Light Theme keyboard shortcut toggle (shift+T)';
+
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isChecked();
+    userMenu.open();
+    userMenu.checkOpen();
+    userMenu.clickMenuLink('Log Out');
+    cy.login();
+    PagePo.goTo(endpoint);
+    prefPage.checkbox(checkboxLabel).isChecked();
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isUnchecked();
+  });
+
+  it('Can select Hide All Type Description Boxes', () => {
+    /*
+    Select the checkbox and verify description banner hidden
+    Deselect the checkbox and verify description banner displays
+    */
+    const banners = new BannersPo();
+    const clusterRepoEndpoint = 'c/_/manager/catalog.cattle.io.clusterrepo';
+    const checkboxLabel = 'Hide All Type Descriptions';
+
+    PagePo.goTo(endpoint);
+    prefPage.checkbox(checkboxLabel).set();
+    prefPage.checkbox(checkboxLabel).isChecked();
+
+    cy.intercept('GET', '/v1/catalog.cattle.io.clusterrepos').as('clusterRepos');
```

Comment:
As we have a ClusterManagerHelmRepoList that extends Page, we can just use `.waitForPage` instead

Reply:
The test fails due to a race condition when I use  `.waitForPage`. When using that function, we're not actually waiting for the page to load full (its only asserting that the url includes the expected endpoint). The intercept logic here actually waits for the GET request's response to be returned before moving on to the next line of code which makes the test more reliable and less prone to failure. Let me know what you think

Reply:
I don't think we have anything generic yet to determine if a page content has loaded. You could do something with `checkVisible` on the first component that you need to do something with though, or i think the getMenuItem might work as well. I think those along with the waitForPage would work.

Have an interceptor with a http request is solid, but it's custom code that would be replicated everywhere. So an alternative solution would be to add the following to `BaseResourceList` 

```
    const repoListPage = new ReposListPagePo('_'); // This can go at the start of the spec file
...
    repoListPage.waitForPage();
    repoListPage.waitForGoTo('/v1/catalog.cattle.io.clusterrepos')
...
  waitForGoTo(validationUrl: string) {
    cy.intercept('GET', validationUrl).as('validateGoTo');
    this.goTo();
    cy.wait('@validateGoTo').its('response.statusCode').should('eq', 200);
  }
```

Reply:
updated. thanks for the suggestions! added `waitForGoTo` to `ReposListPagePo` as discussed.